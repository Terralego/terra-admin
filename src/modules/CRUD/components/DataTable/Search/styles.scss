.dataTable-search {
  margin: 0;
}
