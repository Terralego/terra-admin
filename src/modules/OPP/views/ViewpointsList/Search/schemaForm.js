import {
  TYPE_MANY,
  TYPE_RANGE,
  TYPE_SINGLE,
} from 'mc-tf-test/modules/Forms/Filters';

export const schemaSimpleSearch = [
  {
    property: 'communes',
    label: 'Commune',
    type: TYPE_SINGLE,
    placeholder: 'Le lamentin...',
    values: ['Le lamentin', 'Case-Pilote'],
  }, {
    property: 'themes[]',
    label: 'Thèmes',
    type: TYPE_MANY,
    placeholder: 'Montagnes...',
    display: 'select',
    values: ['Montagnes', 'Éolienne'],
  }, {
    property: 'viewpointDate',
    label: 'Date',
    format: 'date',
    className: 'date_range',
    shortcuts: false,
    contiguousCalendarMonths: false,
    allowSingleDayRange: true,
    startInputProps: {
      placeholder: 'JJ/MM/AAAA',
      className: 'input-range input-range--start',
    },
    endInputProps: {
      placeholder: 'JJ/MM/AAAA',
      className: 'input-range input-range--end',
    },
    type: TYPE_RANGE,
  },
];

export const schemaAdvancedSearch = [
  {
    property: 'communes',
    label: 'Commune',
    type: TYPE_SINGLE,
    placeholder: 'Le lamentin...',
    values: ['Le lamentin', 'Case-Pilote'],

  }, {
    property: 'themes[]',
    label: 'Thèmes',
    type: TYPE_MANY,
    placeholder: 'Montagnes...',
    display: 'select',
    values: ['Montagnes', 'Éolienne'],
  }, {
    property: 'viewpointDate',
    label: 'Date',
    format: 'date',
    className: 'date_range',
    shortcuts: false,
    contiguousCalendarMonths: false,
    allowSingleDayRange: true,
    startInputProps: {
      placeholder: 'JJ/MM/AAAA',
      className: 'input-range input-range--start',
    },
    endInputProps: {
      placeholder: 'JJ/MM/AAAA',
      className: 'input-range input-range--end',
    },
    type: TYPE_RANGE,
  }, {
    property: 'keywords[]',
    label: 'Mots clés',
    type: TYPE_MANY,
    placeholder: 'Filtres...',
    display: 'select',
    values: ['Montagnes', 'Éolienne'],
  },
  {
    property: 'typology[]',
    label: 'Typologie des paysages',
    type: TYPE_MANY,
    placeholder: 'Filtres...',
    display: 'select',
    values: ['Montagnes', 'Éolienne'],
  },
  {
    property: 'photographer',
    label: 'Photographe',
    type: TYPE_SINGLE,
    values: ['Montagnes', 'Éolienne'],
  },
  {
    property: 'label',
    label: 'Nom du point de vue',
    type: TYPE_SINGLE,
    values: ['Montagnes', 'Éolienne'],
  },
  {
    property: 'pictures__id',
    label: 'Id de la Photographie',
    type: TYPE_SINGLE,
    values: ['148', '59', '9992'],
  },
];

export default { schemaSimpleSearch, schemaAdvancedSearch };
