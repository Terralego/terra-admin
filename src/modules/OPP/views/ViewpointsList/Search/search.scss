.search {
  width: 300px;
  flex: none;
  color: #FFF;
  font-size: 0.9em;
  background-color: #293742;
  box-shadow: 0 0 4px 0 rgba(0, 0, 0, 1);

  // Override Blueprintjs Tabs
  &.bp3-tabs {
    background-color: #293742;
    color: #FFF;

    .bp3-tab {
      text-align: center;
      color: #FFF;
      font-family: 'Roboto', sans-serif;
      text-transform: uppercase;
      font-size: 12px;
      width: 50%;
      margin: 0;
      height: 45px;
      padding-top: 10px;
      align-items: baseline;

      &-list {
        background: rgba(255, 255, 255, 0.1);
        justify-content: space-around;
      }

      &-panel {
        margin-top: 0;
      }

      &:hover {
        color: #FFF;
      }

      &-indicator-wrapper {
        width: 50% !important;
        height: 45px !important;
        z-index: 2;

        .bp3-tab-indicator {
          background-color: #FFF;
          width: 50%;
          margin-left: 25%;
        }
      }

      &[aria-selected="true"] {
        transition: all 200ms cubic-bezier(0.4, 1, 0.75, 0.9);
      }
    }
  }
}

