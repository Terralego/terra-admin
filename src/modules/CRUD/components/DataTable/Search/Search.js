import React, { useEffect, useState, useRef } from 'react';
import PropTypes from 'prop-types';
import debounce from 'lodash.debounce';
import { Icon, InputGroup, Spinner } from '@blueprintjs/core';
import './styles.scss';

const CustomLoader = ({ isSearching }) => (
  isSearching
    ? <Spinner size={Icon.SIZE_STANDARD} />
    : <div style={{ width: 30 }} />
);

const Search = ({
  featureEndpoint,
  featuresList,
  getFeaturesList,
  setTableFilters,
  t,
  tableFilters,
}) => {
  const { search: defaultValue } = tableFilters;
  const inputRef = useRef(null);
  const [isSearching, setIsSearching] = useState(false);

  useEffect(() => {
    inputRef.current && inputRef.current.focus();
    setIsSearching(false);
  }, [featuresList]);

  const loadData = async value => {
    const querystring = { ...tableFilters, page: 1, search: value };
    await getFeaturesList(featureEndpoint, querystring);
    setTableFilters(querystring);
  };

  const delayedQuery = useRef(
    debounce(value => {
      if (!value.length || value.length > 2) {
        setIsSearching(true);
        if (featureEndpoint) {
          loadData(value);
        }
      }
    },
    500),
  ).current;

  const onChange = ({ target: { value } }) => {
    delayedQuery(value);
  };

  return (
    <InputGroup
      defaultValue={defaultValue}
      inputRef={inputRef}
      onChange={onChange}
      minimal
      leftIcon="search"
      rightElement={<CustomLoader isSearching={isSearching} />}
      disabled={isSearching}
      title={t('CRUD.table.search')}
      type="search"
    />
  );
};

Search.propTypes = {
  featureEndpoint: PropTypes.string,
  featuresList: PropTypes.shape({}),
  getFeaturesList: PropTypes.func,
  setTableFilters: PropTypes.func,
  t: PropTypes.func,
  tableFilters: PropTypes.shape({}),
};

Search.defaultProps = {
  featureEndpoint: undefined,
  featuresList: {},
  getFeaturesList: () => {},
  setTableFilters: () => {},
  t: () => {},
  tableFilters: {},
};

export default Search;
