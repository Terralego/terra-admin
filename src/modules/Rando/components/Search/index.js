import React, { useState } from 'react';
import { InputGroup, Button } from '@blueprintjs/core';

import './styles.scss';

const Search = () => {
  const [query, setQuery] = useState('');
  return (
    <div className="rando-search">
      <InputGroup
        leftIcon="filter"
        value={query}
        onChange={({ target: { value } }) => setQuery(value)}
        rightElement={(
          <Button
            icon="cross"
            onClick={() => setQuery('')}
            minimal
          />
        )}
      />
    </div>
  );
};

export default Search;
