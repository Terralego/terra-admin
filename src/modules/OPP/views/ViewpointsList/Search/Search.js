import React from 'react';

import { Tab, Tabs } from '@blueprintjs/core';
import i18n from 'i18next';

import { schemaSimpleSearch, schemaAdvancedSearch } from './schemaForm';
import SearchForm from './SearchForm';

import './search.scss';
import { fetchValuesToFiltering } from '../../../services/viewpoints';

const ID_SEARCH_PANEL = 'search';
const ID_SEARCH_SIMPLE = 'searchSimple_tab';
const ID_SEARCH_ADVANCED = 'searchAdvanced_tab';

const locales = {
  noResults: i18n.t('opp.viewpoint.filter.no-result'),
  overlappingDatesMessage: i18n.t('date.error.overlapping-date'),
  invalidDateMessage: i18n.t('date.error.invalid-date'),
  emptySelectItem: i18n.t('opp.viewpoint.filter.empty-selected-input'),
};

export class Search extends React.Component {
  state = {
    navTabId: ID_SEARCH_SIMPLE,
    filtersOfSimpleSearch: [],
    filtersOfAdvancedSearch: [],
  };

  componentDidMount () {
    this.getFilters();
  }

  handleNavSearchTabChange = navTabId => this.setState({ navTabId });

  getValuesFilters = (schemaForm, data) => {
    const keys = Object.keys(data);

    return schemaForm.map(filter => {
      if (keys.indexOf(filter.name) > -1) {
        return { ...filter, ...{ values: data[filter.name] } };
      }
      return { ...filter };
    });
  };

  getFilters = async () => {
    try {
      const data = await fetchValuesToFiltering();
      const filtersOfSimpleSearch = this.getValuesFilters(schemaSimpleSearch, data);
      const filtersOfAdvancedSearch = this.getValuesFilters(schemaAdvancedSearch, data);

      this.setState({ filtersOfSimpleSearch, filtersOfAdvancedSearch });
    } catch (e) {
      // eslint-disable-next-line no-console
      console.error(e);
    }
  };

  render () {
    const {
      navTabId,
      filtersOfSimpleSearch,
      filtersOfAdvancedSearch,
    } = this.state;
    const {
      itemsPerPage,
      t,
      handleResetPage,
    } = this.props;

    return (
      <Tabs
        id={ID_SEARCH_PANEL}
        onChange={this.handleNavSearchTabChange}
        selectedTabId={navTabId}
        className="search"
      >
        <Tab
          id={ID_SEARCH_SIMPLE}
          className="search-filters search-simple-content"
          title={t('opp.viewpoint.filter.simple-search')}
          panel={(
            <SearchForm
              itemsPerPage={itemsPerPage}
              filters={filtersOfSimpleSearch}
              locales={locales}
              handleResetPage={handleResetPage}
            />
          )}
        />

        <Tab
          id={ID_SEARCH_ADVANCED}
          className="search-filters search-advanced-content"
          title={t('opp.viewpoint.filter.advanced-search')}
          panel={(
            <SearchForm
              itemsPerPage={itemsPerPage}
              filters={filtersOfAdvancedSearch}
              locales={locales}
              handleResetPage={handleResetPage}
            />
          )}
        />
      </Tabs>
    );
  }
}

export default Search;
