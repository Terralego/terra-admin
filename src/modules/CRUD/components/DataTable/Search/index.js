import { withTranslation } from 'react-i18next';
import { withRouter } from 'react-router-dom';
import { connectCRUDProvider } from '../../../services/CRUDProvider';
import { withTableFilters } from '../../../services/UserSettingsProvider';
import { getView } from '../../../services/CRUD';
import compose from '../../../../../utils/compose';

import Search from './Search';

const CRUDPRoviderGetter = ({
  featuresList,
  getFeaturesList,
  settings,
}, {
  match: { params: { layer } },
}) => {
  const { featureEndpoint } = getView(settings, layer);
  return {
    featureEndpoint,
    featuresList,
    getFeaturesList,
  };
};

export default compose(
  withRouter,
  connectCRUDProvider(CRUDPRoviderGetter),
  withTableFilters(),
  withTranslation(),
)(Search);
