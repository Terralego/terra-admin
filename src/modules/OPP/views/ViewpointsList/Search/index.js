import { withTranslation } from 'react-i18next';

import { Search } from './Search';

export default (withTranslation()(Search));
