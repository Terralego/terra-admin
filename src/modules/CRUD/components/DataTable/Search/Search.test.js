import React from 'react';
import renderer from 'react-test-renderer';

import Search from './Search';

jest.mock('@blueprintjs/core', () => ({
  InputGroup: ({ children, ...props }) => <div {...props}>{children}</div>,
  Icon: {
    SIZE_STANDARD: 'standard',
  },
  Spinner: () => <div>Loading</div>,
}));

jest.mock('lodash.debounce', props => fn => () => fn(props));

let props;
beforeEach(() => {
  props = {
    loadData: () => {},
    queryString: {},
  };
});

it('should render correctly', () => {
  const tree = renderer.create((
    <Search
      {...props}
    />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render correctly with default value', () => {
  const tree = renderer.create((
    <Search
      {...props}
      tableFilters={{ search: 'foo' }}
    />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
