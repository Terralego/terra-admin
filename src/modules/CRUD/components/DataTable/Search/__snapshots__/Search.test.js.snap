// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  disabled={false}
  inputRef={
    Object {
      "current": null,
    }
  }
  leftIcon="search"
  minimal={true}
  onChange={[Function]}
  rightElement={
    <CustomLoader
      isSearching={false}
    />
  }
  type="search"
/>
`;

exports[`should render correctly with default value 1`] = `
<div
  defaultValue="foo"
  disabled={false}
  inputRef={
    Object {
      "current": null,
    }
  }
  leftIcon="search"
  minimal={true}
  onChange={[Function]}
  rightElement={
    <CustomLoader
      isSearching={false}
    />
  }
  type="search"
/>
`;
